### PR TITLE
Camera acquisitor system

### PR DESCRIPTION
**Pull request by X52p**
(needs vector pointing first)

### CODE REVIEW

File: core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml [hunk 1]
```diff
@@ -195,6 +195,60 @@
...
+            <mal:errorRef
+              comment="The Camera unit is not available.">
+              <mal:type area="Platform" list="false"
+                        name="DEVICE_NOT_AVAILABLE" />
+            </mal:errorRef>
+            <mal:errorRef
+              comment="The requested resolution is not supported.">
+              <mal:type area="COM" list="false" name="INVALID" />
+              <mal:extraInformation
+                comment="The extra information fied shall hold the list of available resolutions.">
```
Comment (wbnns) [inline core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml:235]: s/fied/field/

File: core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml [hunk 2]
```diff
@@ -195,6 +195,60 @@
...
+                <mal:type area="Platform" list="true"
+                          name="PixelResolution" service="Camera" />
+              </mal:extraInformation>
+            </mal:errorRef>
+            <mal:errorRef
+              comment="The requested format is not supported.">
+              <mal:type area="COM" list="false" name="INVALID" />
+              <mal:extraInformation
+                comment="The extra information fied shall hold the list of available formats.">
```
Comment (wbnns) [inline core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml:244]: s/fied/field/

File: sdk/examples/ground/camera-acquisitor-system-ground/src/main/java/esa/mo/ground/cameraacquisotorground/SpringInitiator.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  /**
+   * Main command line entry point.
+   *
+   * @param args the command line arguments
+   */
+  public static void main(final String args[])
+  {
+    if (args.length != 2) {
+      System.err.println("Please give directoryURI as first argument and Port as seconds argument!");
```
Comment (wbnns) [inline sdk/examples/ground/camera-acquisitor-system-ground/src/main/java/esa/mo/ground/cameraacquisotorground/SpringInitiator.java:43]: s/seconds/second/

File: sdk/examples/ground/camera-acquisitor-system-ground/src/main/java/esa/mo/ground/restservice/Pass.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    } else {
+      this.passEnd = s.getDate();
+      double elepsedTime = this.passEnd.durationFrom(this.passStart);
+      this.optimalTime = this.passStart.shiftedBy(elepsedTime / 2);
+      this.resultTime = this.optimalTime.toString();
+      if (this.optimalTime.durationFrom(CameraAcquisitorSystemMCAdapter.getNow()) <= worstCaseRotationTimeSeconds && this.optimalTime.compareTo(
+          this.notBeforeDate) > 0) {
+
+        //time too close, try again
```
Comment (wbnns) [inline sdk/examples/ground/camera-acquisitor-system-ground/src/main/java/esa/mo/ground/restservice/Pass.java:98]: Missing space between `//` and `/time`

File: sdk/examples/space/camera-acquisitor-system/pom.xml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    <artifactId>parent</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+    <relativePath>../../../../parent/pom.xml</relativePath>
+  </parent>
+  <groupId>int.esa.nmf.sdk.examples.space</groupId>
+  <artifactId>camera-acquisitor-system</artifactId>
+  <packaging>jar</packaging>
+
+  <name>ESA NMF SDK App Example - Camera acquisitor system</name>
+  <description>An app for planning camera images based on the satelites position</description>
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/pom.xml:26]: s/satelites/satellite's/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraHandler.java [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+    }
+  }
+
+  /**
+   * Takes a photograph (instantly)
+   *
+   * @param actionInstanceObjId the Instance ID of the action that triggers the Photograph
+   * @param stageOffset         number of states that where already executed before taking a
+   *                            photograph
+   * @param totalStages         total number of Stages (including all stages needet for a
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraHandler.java:280]: s/needet/needed/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraTargetHandler.java [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+  private static final int STAGE_WAIT_FOR_PASS = 1;
+  private static final int STAGE_ATTITUDE_CORECTION = 2;
+  private static final int STAGE_WAIT_FOR_OPTIMAL_PASS = 3;
+
+  private final CameraAcquisitorSystemMCAdapter casMCAdapter;
+
+  private static final Logger LOGGER = Logger.getLogger(
+      CameraAcquisitorSystemCameraTargetHandler.class.getName());
+
+  // time to keep track of all sheduled photographs
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraTargetHandler.java:72]: s/sheduled/scheduled/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraTargetHandler.java [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+
+  UInteger photographLocation(double longitude, double latitude, String timeStemp,
+      Long actionInstanceObjId,
+      boolean reportProgress, MALInteraction interaction)
+  {
+    AbsoluteDate targetDate = new AbsoluteDate(timeStemp, TimeScalesFactory.getUTC());
+
+    double seconds = targetDate.durationFrom(CameraAcquisitorSystemMCAdapter.getNow());
+
+    // create TimerTask to shedule attitude correction shortly before targetDate
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemCameraTargetHandler.java:104]: s/shedule/schedule/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+      generationEnabled = false)
+  private byte exposureType = 0;
+
+  @Parameter(
+      description = "The camera's exposure time (only used if exposureType is CUSTOM)",
+      generationEnabled = false)
+  private float exposureTime = 1.0f;
+
+  @Parameter(description =
+      "The maximum time (in Milliseconds) the Satelite will take to rotated if its in the worst posible orientation",
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java:92]: /posible/possible/

Comment (wbnns) [reply sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java:92]: s/its/it's/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+      generationEnabled = false)
+  private long worstCaseRotationTimeMS = 1000000;
+
+  @Parameter(description =
+      "The time (in Milliseconds) the Satelite will start the attitude control prior to reaching a target",
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java:97]: s/Satelite/Satellite/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+  public NMFInterface getConnector()
+  {
+    return connector;
+  }
+
+  public CameraAcquisitorSystemMCAdapter(final NMFInterface connector)
+  {
+    try {
+      //load orekit-data wich is required for many parts of orekit to work.
```
Comment (wbnns) [inline sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java:190]: Missing space between `//` and `load`

File: sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestCameraDataHandler.java [hunk 12]
```diff
@@ -68,12 +72,19 @@ public void takePictureAckReceived(
...
+    takePictureAckReceived(msgHeader, qosProperties);
+  }
+
   @Override
   public void takePictureResponseReceived(
       org.ccsds.moims.mo.mal.transport.MALMessageHeader msgHeader,
       org.ccsds.moims.mo.platform.camera.structures.Picture picture,
       java.util.Map qosProperties)
   {
+    LOGGER.log(Level.INFO, "Image has been recived");
```
Comment (wbnns) [inline sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestCameraDataHandler.java:87]: s/recived/received/

File: sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestCameraDataHandler.java [hunk 13]
```diff
@@ -162,4 +192,10 @@ public void takePictureResponseErrorReceived(
     }
   }
 
+  @Override
+  public void takeAutoExposedPictureResponseErrorReceived(MALMessageHeader msgHeader,
+      MALStandardError error, Map qosProperties)
+  {
+    takePictureResponseErrorReceived(msgHeader, error, qosProperties); //To change body of generated methods, choose Tools | Templates.
```
Comment (wbnns) [inline sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestCameraDataHandler.java:199]: Missing space between `//` and `To`

File: sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java [hunk 14]
```diff
@@ -524,6 +524,40 @@ public UInteger takePicture_BMP(
         .takePicture(actionInstanceObjId, reportProgress, interaction, PictureFormat.BMP);
   }
 
+  @Action(description = "Uses the NMF Camera service to take a auto exposed picture in RAW format.",
```
Comment (wbnns) [inline sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java:527]: s/a/an/

File: sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java [hunk 15]
```diff
@@ -524,6 +524,40 @@ public UInteger takePicture_BMP(
...
+  public UInteger takeAutoExposedPicture_RAW(
+      Long actionInstanceObjId,
+      boolean reportProgress,
+      MALInteraction interaction)
+  {
+    return actionsHandler
+        .takeAutoExposedPicture(actionInstanceObjId, reportProgress, interaction, PictureFormat.RAW);
+  }
+
+  @Action(description = "Uses the NMF Camera service to take a auto exposed picture in JPG format.",
```
Comment (wbnns) [inline sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java:538]: s/a/an/

Comment (wbnns) [reply sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java:538]: s/JPG/RAW/

File: sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java [hunk 16]
```diff
@@ -524,6 +524,40 @@ public UInteger takePicture_BMP(
...
+      Long actionInstanceObjId,
+      boolean reportProgress,
+      MALInteraction interaction)
+  {
+    return actionsHandler
+        .takeAutoExposedPicture(actionInstanceObjId, reportProgress, interaction, PictureFormat.JPG);
+  }
+
+
+  @Action(description = "Uses the NMF Camera service to take a auto exposed picture in BMP format.",
```
Comment (wbnns) [inline sdk/examples/space/payloads-test/src/main/java/esa/mo/nmf/apps/PayloadsTestMCAdapter.java:550]: s/a/an/

File: sdk/examples/ground/camera-acquisitor-system-ground/pom.xml [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+    <version>2.0.0-SNAPSHOT</version>
+    <relativePath>../../../../parent/pom.xml</relativePath>
+  </parent>
+
+  <groupId>int.esa.nmf.sdk.examples.ground</groupId>
+  <artifactId>camera-acquisitor-system-ground</artifactId>
+  <packaging>jar</packaging>
+
+  <name>ESA NMF SDK Ground Example - Camera Acquisitor System</name>
+  <description>A advanced demo for the Ground part, setting a parameter and submitting action to a provider exposing a REST API</description>
```
Comment (wbnns) [inline sdk/examples/ground/camera-acquisitor-system-ground/pom.xml:30]: s/A advanced/An advanced/

File: docs/source/apps/apps.rst [hunk 18]
```diff
@@ -25,6 +25,17 @@ The recommended way to create a new NMF App project is to copy one of the exampl
 So in our case, we will copy the camera example project folder and rename it to something more fitting, e.g. 'sobel'.
 To make our life easier when we import the project into our workspace, we should edit the app's POM to give it a unique name and artifact ID.
 So in the ``pom.xml`` inside the folder **sobel** change the field ``artifactId`` from the value 'camera' to 'sobel'. You should also change the values of the ``name`` and ``description`` tags to relate to our example and put your name into the ``author`` tags. 
+If you need to use OreKit in your App and don't want to use custom orekit-data, than you sould alsow add the following dependency:
```
Comment (wbnns) [inline docs/source/apps/apps.rst:28]: s/than/then/

Comment (wbnns) [reply docs/source/apps/apps.rst:28]: s/sould/should/

Comment (wbnns) [reply docs/source/apps/apps.rst:28]: s/alsow/also/

File: sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java (hunk 9 above)
Comment (wbnns) [reply sdk/examples/space/camera-acquisitor-system/src/main/java/esa/mo/nmf/apps/CameraAcquisitorSystemMCAdapter.java:92]: s/Satelite/Satellite/

File: docs/source/apps/apps.rst (hunk 18 above)
Comment (CesarCoelho) [reply docs/source/apps/apps.rst:28]: @dmarszk 
I suggest to copy the content in each XML file and drop it in Word. Then you will immediately see all the typos and mistakes! Should be faster that way!

**Comment by dmarszk** [comment]
Already merged in other PR #24 - this only covers some typo fixes.

**Pull request merged**